solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/norm/impl/StandardNormalizer.java

```diff
@@ -51,24 +52,25 @@ public void setStd(String std) {
   }
 
   @Override
-  public void init(NamedParams params) throws NormalizerException {
-    super.init(params);
-    if (!params.containsKey("avg")) {
-      throw new NormalizerException("missing param avg");
-    }
-    if (!params.containsKey("std")) {
-      throw new NormalizerException("missing param std");
-    }
-    avg = params.getFloat("avg", 0);
-    std = params.getFloat("std", 1);
-    if (std <= 0) {
-      throw new NormalizerException("std must be > 0");
-    }
+  public float normalize(float value) {
+    return (value - avg) / std;
```

Comment:
Yes, invalid parameters would result in an exception at query time. Exceptions at query time could happen anyhow e.g. by asking to use a non-existing model and so attempting to use an existing but invalidly configured model would be similar.

Alternatively the setters could be throwing (Normalizer)Exception exceptions but then that would need to be caught and handled further up the calling chain i.e. extra code/complexity when it should be relatively unlikely/difficult to get the parameters wrong. Notice that it is now possible and okay to not configure min/max/avg/std parameters i.e. in the absence of configuration valid defaults would be used.